src/audio/module_adapter/module_adapter.c

```diff
@@ -1069,32 +1069,28 @@ static int module_adapter_copy_dp_queues(struct comp_dev *dev)
 		dp_queue = dp_queue_get_next_item(dp_queue);
 	}
 
+	if (mod->dp_startup_delay)
+		return 0;
+
 	dp_queue = dp_queue_get_first_item(&mod->dp_queue_dp_to_ll_list);
 	list_for_item(blist, &dev->bsink_list) {
-		/* output - we need to copy data from dp_queue (as source)
-		 * to audio_stream (as sink)
-		 *
-		 * a trick is needed there as a workaround
-		 * DP may produce a huge chunk of output data (i.e. 10 LL
-		 * cycles), and the following module should be able to consume it in 1 cycle chunks
-		 *
-		 * unfortunately some modules are not prepared to work when there's more than
-		 * 1 data portion available in the buffer and are draining buffers with data loss
-		 *
-		 * a workaround: copy only the following module's IBS in each LL cycle
-		 *
-		 * required fix: all modules using sink/src interface must be aware to
-		 * process only data they need, not forcefully draining a buffer
-		 */
+	/* output - we need to copy data from dp_queue (as source)
+	 * to audio_stream (as sink)
+	 */
```

Comment:
I don't think we should drop this comment (from first patch in the series). The logic to not feed more than on LL is still in place, right, and should be explained.